pkg/aws/wait.go

```diff
@@ -159,3 +161,23 @@ func waitForClusterOperators(cfg *rest.Config) error {
 	_, err = clientwatch.UntilWithSync(ctx, listWatcher, &configapi.ClusterOperator{}, nil, clusterOperatorsAreAvailable)
 	return err
 }
+
+func waitForServiceLoadBalancerDNS(client kubeclient.Interface, namespace, name string) (string, error) {
+	var serviceDNS string
+
+	err := wait.PollImmediate(10*time.Second, serviceLoadBalancerTimeout, func() (bool, error) {
+		svc, err := client.CoreV1().Services(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		if len(svc.Status.LoadBalancer.Ingress) < 1 {
+			return false, nil
+		}
+
+		serviceDNS = svc.Status.LoadBalancer.Ingress[0].Hostname
```

Comment:
Return either `svc.Status.LoadBalancer.Ingress[0].Hostname` or `svc.Status.LoadBalancer.Ingress[0].IP` whichever is not blank. In Azure and GCP, you'll get the IP.

Reply:
Also in the future we may want to rename the param `ExternalAPIAddress` instead of `ExternalAPIDNSName`